toolchain/parser/parse_node_kind.h

```diff
@@ -64,6 +67,16 @@ class ParseNodeKind {
 
   void Print(llvm::raw_ostream& out) const { out << name(); }
 
+  // Returns true if the node is bracketed; otherwise, child_count is used.
+  auto has_bracket() const -> bool;
+
+  // Returns the bracketing character for the node. Requires that has_bracket is
+  // true.
+  auto bracket() const -> ParseNodeKind;
+
+  // Returns the child_count for the node. Requires that has_bracket is false.
```

Comment:
Might be worth some description of what this means? Maybe can even refer to it from the x-macro file if you want.

Reply:
Iterated on comments.